File: backend/app/schemas/artifact.py
"""Artifact schemas."""

from datetime import datetime
from pydantic import BaseModel, ConfigDict


class ArtifactBase(BaseModel):
    """Base artifact schema."""
    filename: str
    artifact_type: str | None = None
    description: str | None = None


class ArtifactCreate(ArtifactBase):
    """Schema for creating artifact metadata (file upload handled separately)."""
    run_id: int


class ArtifactResponse(ArtifactBase):
    """Schema for artifact response."""
    id: int
    content_type: str | None = None
    size_bytes: int
    sha256_hash: str
    run_id: int
    uploaded_by_id: int
    is_deleted: bool
    created_at: datetime

    model_config = ConfigDict(from_attributes=True)


class ArtifactUploadResponse(BaseModel):
    """Schema for artifact upload response."""
    artifact: ArtifactResponse
    message: str = "Artifact uploaded successfully"

File: frontend-react/src/types/artifact.ts
// 附件相关类型，对应后端 schemas/artifact.py。
// 附件（Artifact）是工单下上传的文件，如路由器配置文件。

export interface Artifact {
  id: number
  filename: string
  artifact_type: string | null
  description: string | null
  content_type: string | null
  size_bytes: number
  sha256_hash: string
  ticket_id: number
  run_id: number | null
  uploaded_by_id: number
  is_deleted: boolean
  created_at: string
}
